File: Examples/VisualBasic/PrintingAndExporting/SupportForSQLiteDatabase.vb
Imports System.Collections.Generic
Imports System.IO
Imports Aspose.Tasks
Imports Aspose.Tasks.Connectivity
Imports Aspose.Tasks.Saving
Imports Aspose.Tasks.Visualization

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Tasks.Examples.VisualBasic.PrintingAndExporting
    Public Class SupportForSQLiteDatabase
        Public Shared Sub Run()
            ' ExStart:SupportForSQLiteDatabase
            ' The path to the documents directory.
            Const connectionString As String = "SQLiteDatabaseConnectionString"
            Const projectId As Integer = 4502
            Dim primaveraDbSettings As New PrimaveraDbSettings(connectionString, projectId)
            primaveraDbSettings.ProviderInvariantName = "System.Data.SQLite"
            Dim project As New Project(primaveraDbSettings)
            ' ExEnd:SupportForSQLiteDatabase
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithProjects/ImportingAndExporting/SupportForSQLiteDatabase.cs
/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from https://www.nuget.org/packages/Aspose.Tasks/, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using https://forum.aspose.com/c/tasks
*/

namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects.ImportingAndExporting
{
    using System;

    using Aspose.Tasks.Connectivity;
    using Aspose.Tasks.Saving;

    internal class SupportForSQLiteDatabase
    {
        public static void Run()
        {
             // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            try
            {
                //ExStart:SupportForSQLiteDatabase
                var connectionString = "Data Source=" + dataDir + "\\PPMDBSQLite.db";
                const int ProjectId = 4502;

                // Create Primavera DB Settings using connection string and project id
                var settings = new PrimaveraDbSettings(connectionString, ProjectId);
                settings.ProviderInvariantName = "System.Data.SQLite";

                // Create new project using primavera db settings
                var project = new Project(settings);
                project.Save(dataDir + "SupportForSQLiteDatabase_out.mpp", SaveFileFormat.MPP);
                //ExEnd:SupportForSQLiteDatabase
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message + "\nPlease setup proper data source (connectionString, ProviderInvariantName) etc");      
            }            
        }
    }
}